src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -164,27 +171,43 @@ public class CoinstakeContext
         /// but high enough to compensate for tasks' overhead.</remarks>
         private const int UtxoStakeDescriptionsPerCoinstakeWorker = 25;
 
+        /// <summary>Responsible for downloading and validating blocks.</summary>
```

Comment:
puller is responsible for downloading blocks

consensus loop is a component that consumes incoming blocks from puller or from miner/staker and then validates and executes (processes) such a block. processing means that it modifies the UTXO set according to the transactions in the block (among other things)